### PR TITLE
fixes for 12.1 windows compiler

### PR DESCRIPTION
**Pull request by earonesty**
had to make these changes to get visual studio to compile deepspeed as well as some changes to the torch cpp_extension.py (which didn't include c10 lib in the linker args).  all the torch libs use int64_t and not some special torch_long_int or somethng.  imo, it sholuld probably be the latter.

### CODE REVIEW

**Comment by earonesty** [comment]
> @microsoft-github-policy-service agree

@microsoft-github-policy-service agree

File: csrc/quantization/pt_binding.cpp [hunk 1]
```diff
@@ -9,6 +9,7 @@
 #include <vector>
 #include "quantization.h"
 
+
```
Comment (loadams) [inline csrc/quantization/pt_binding.cpp:12]: You'll want to remove this additional new line to fir our formatting guidelines

File: csrc/quantization/pt_binding.cpp [hunk 2]
```diff
@@ -192,16 +192,16 @@ std::vector<at::Tensor> quantized_reduction(at::Tensor& input_vals,
                               .layout(at::kStrided)
                               .device(at::kCUDA)
                               .requires_grad(false);
-    const int scales_elems = (quantize::requires_offset(quant_type)) ? 2 : 1;
-    auto scales = torch::empty({out_groups, scales_elems}, scales_options);
+    const int64_t scales_elems = (quantize::requires_offset(quant_type)) ? 2 : 1;
+    auto scales = torch::empty({(int64_t)out_groups, scales_elems}, scales_options);
```
Comment (loadams) [inline csrc/quantization/pt_binding.cpp:196]: @earonesty - could you give samples of the errors you were hitting that caused you to change these?

Comment (loadams) [reply csrc/quantization/pt_binding.cpp:196]: And was this to build all ops other than those not suppored on Windows (aio)?

Comment (earonesty) [reply csrc/quantization/pt_binding.cpp:196]: yes, this was everything i needed.   it was mostly "template ambiguity" errors.   there are multiple torch::empty funcs in the header. i can remove them and get the exact errors.

Comment (earonesty) [reply csrc/quantization/pt_binding.cpp:196]: that being said, code should probably always use int64_t when interacting with torch tensors, as a general rule

**Pull request closed without merging**